### PR TITLE
collect freshness fix

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
**What change does this PR introduce?** 
 
Fivetran created PR

**If this PR introduces a new macro, how did you test the new macro?** 

This PR adjusts the `collect_freshness` macro to now refer to the `identifier` instead of the `name` of the node in the src.yml. This ensures if a dbt user has the `identifier` field populated, then the macro will work as intended. This has a high impact since this macro affects non Fivetran dbt packages. We will want to ensure this approach is a solid fix before merging. 

Additionally, we will most likely want to take a different approach to disabling freshness tests in our packages. The current solution is a too invasive and impacts parts of dbt projects that are outside the scope of Fivetran packages. We should use [this](https://github.com/dbt-labs/dbt-core/issues/4525) issue as inspiration for determining alternative approaches to freshness tests.

**If this PR introduces a modification to an existing macro, which packages is the macro currently present in and what steps were taken to test compatibility across packages?** 

All source freshness tests that exist in users dbt projects if they have a Fivetran dbt package installed.

**Did you update the README to reflect the macro addition/modifications?** 

- [ ] Yes
- [X] No (provide further explanation)

This is an under the hood update and does not need a README update.

### CODE REVIEW

File: macros/collect_freshness.sql [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
   {%- set enabled_array = [] -%}
   {% for node in graph.sources.values() %}
-    {% if node.name == source.name %}
+    {% if node.identifier == source.identifier %}
```
Comment (fivetran-jamie) [inline macros/collect_freshness.sql:11]: if an identifier or alias isn't explicitly provided to a source, is `identifier` the `name` by default? @fivetran-joemarkiewicz

Comment (fivetran-joemarkiewicz) [reply macros/collect_freshness.sql:11]: That is what I got from my testing. When using this on nodes with **only** the `name` config, it just defaulted to use the name.

Comment (fivetran-jamie) [reply macros/collect_freshness.sql:11]: ah nice seems like it does

File: macros/collect_freshness.sql [hunk 2]
```diff
@@ -17,10 +17,10 @@
   {% set is_enabled = (enabled_array != []) %}
 
     select
-      {% if is_enabled %}
-      max({{ loaded_at_field }})
+      {% if is_enabled == false %}
```
Comment (fivetran-jamie) [inline macros/collect_freshness.sql:20]: why this change? just wanna make sure there's not an issue with the conditional on line 26 as well

**Review by fivetran-jamie — APPROVED** [review]

**Review by fivetran-jamie — APPROVED** [review]
looks good!
